tools/update_extern.py

```diff
@@ -0,0 +1,116 @@
+"""
+Updates all the libraries in ``sunpy/extern``
+"""
+
+import os
+import shutil
+from zipfile import ZipFile
+import requests
+from pathlib import Path
+
+SUNPY_DIR = Path(__file__).parent.parent
+
+
+PACKAGES = {
+    "appdirs": ["ActiveState", "appdirs.py"],
+    "distro": ["python-distro", "src/distro/distro.py"],
+    "inflect": ["jaraco", "inflect/__init__.py"],
+    "parse": ["r1chardj0n3s", "parse.py"],
+}
```

Comment:
Hi @akash5100, sorry I have just realised in the original issue I forgot to mention that the `modest_image.py` file needs to be updated (as it was buried within the `glue` package). I tried including it in the list of packages although currently the script would have copied it to extern as `glue.py` rather than `modest_image.py`. 

I think the following convention for the dictionary of packages might make more sense going forward, i.e. keys are unique names for the files in the extern directory, and the values are lists of where to find the file: [user, repository, path]. That way it would be possible to update two separate files taken from the same GitHub repository.
```suggestion
PACKAGES = {
    "appdirs": ["ActiveState", "appdirs", "appdirs.py"],
    "distro": ["python-distro", "distro", "src/distro/distro.py"],
    "inflect": ["jaraco", "inflect", "inflect/__init__.py"],
    "modest_image": ["glue-viz", "glue", "glue/external/modest_image.py"],
    "parse": ["r1chardj0n3s", "parse", "parse.py"],
}
```

I do still think we should simplify the script by instead updating one file at a time. That way we wouldn't need to walk through directories to find the file we want to copy across, as we can keep track of everything inside the function based on the return values (see how I got the path to the extracted file in https://github.com/sunpy/sunpy/pull/6127#discussion_r872409761). What I'm thinking is a function:

```py
def download_github_file(user, repo, src, dest):
```

where `user` and `repo` are the GitHub username and repository where the file is present, `src` is the path to the file within the GitHub repository, and `dest` is where to download the file to (e.g. `/Users/username/sunpyrepo/sunpy/extern/name.py`). This function could then be called in a `for` loop for each entry in the `PACKAGES` dictionary as shown in https://github.com/sunpy/sunpy/pull/6127#discussion_r872409761. 

Sorry for the last minute change, although I think this is a great opportunity to create a function that can grab a file from a GitHub repository and place it in any directory you like!! Therefore the function itself could be used by other projects and not just SunPy! 😀

Reply:
I actually think in this case, we can remove `modest_image.py`, I can't find where it is imported anymore. I suspect it was needed when we had the animation source code in sunpy but now we dont.

Reply:
I think modest image was only used by the animators and it's now in `mpl-animators` and I forgot to remove it from here.